tests/deepsparse/loggers/metric_functions/template_functions/test_template_functions.py

```diff
@@ -12,11 +12,148 @@
+        ),
+        (
+            "segmentation",
+            "yolact",
+            {"images": numpy.ones((2, 3, 640, 640))},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/segmentation_logs.txt",  # noqa E501
+        ),
+        (
+            "sentiment_analysis",
+            "sentiment_analysis",
+            {"sequences": "the food tastes great"},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/sentiment_analysis_logs_1.txt",  # noqa E501
+        ),
+        (
+            "sentiment_analysis",
+            "sentiment_analysis",
+            {"sequences": ["the food tastes great", "the food tastes bad"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/sentiment_analysis_logs_2.txt",  # noqa E501
+        ),
+        (
+            "sentiment_analysis",
+            "sentiment_analysis",
+            {
+                "sequences": [
+                    ["the food tastes great"],
+                    ["the food tastes bad"],
+                ]
+            },
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/sentiment_analysis_logs_3.txt",  # noqa E501
+        ),
+        (
+            "zero_shot_text_classification",
+            "zero_shot_text_classification",
+            {"sequences": "the food tastes great", "labels": ["politics", "food"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/zero_shot_text_classification_logs_1.txt",  # noqa E501
+        ),
+        (
+            "zero_shot_text_classification",
+            "zero_shot_text_classification",
+            {
+                "sequences": ["the food tastes great", "the government is corrupt"],
+                "labels": ["politics", "food"],
+            },
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/zero_shot_text_classification_logs_2.txt",  # noqa E501
+        ),
+        (
+            "token_classification",
+            "token_classification",
+            {"inputs": "the food tastes great"},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/token_classification_logs_1.txt",  # noqa E501
+        ),
+        (
+            "token_classification",
+            "token_classification",
+            {"inputs": ["the food tastes great", "the food tastes bad"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/token_classification_logs_2.txt",  # noqa E501
+        ),
+    ],
+)
+@mock_engine(rng_seed=0)
+def test_group_name(mock_engine, group_name, pipeline_name, inputs, expected_logs):
+    yaml_config = """
+    loggers:
+        list_logger:
+            path: tests/deepsparse/loggers/helpers.py:ListLogger
+    add_predefined:
+    - func: {group_name}"""
+
+    if pipeline_name == "zero_shot_text_classification":
```

Comment:
Hack needed to make the code run with the pipeline that does not have `task` property set by default

Reply:
let's make a ticket to get a fix in